offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -66,6 +66,15 @@ export class CkbHandler {
     await this.kvDb.set(lastHandleCkbBlockKey, `${blockNumber},${blockHash}`);
   }
 
+  getHandledBlock(): { height: number; hash: string } {
+    return { height: this.lastHandledBlockHeight, hash: this.lastHandledBlockHash };
+  }
+
+  async getTipBlock(): Promise<{ height: number; hash: string }> {
+    const tipHeader = await this.ckb.rpc.getTipHeader();
+    return { height: Number(tipHeader.number), hash: (this.lastHandledBlockHash = tipHeader.hash) };
```

Comment:
an `getTipBlock` should not cause a state change, state changes should be explicit

```ts
const block = handler.getTipBlock();
handler.setLatestHandledBlockHash*(
```

Reply:
mistake